Refactor scheduler_perf_test codes

kerthcet opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?
/kind cleanup
/sig scheduling

<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

#### What this PR does / why we need it:

#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

#### Special notes for your reviewer:

Three commits here:
1) Add a new definition type for `operationCode`
2) Most about the annotations, make it more readability. Another change is replace if-else with switch-case clause.
3) Rename `SchedulerConfigFile` to `SchedulerConfigFilePath` and make it a pointer to be consistent with `DefaultPodTemplatePath`.

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

kerthcet committed:
Use operationCode instead of string directly

Signed-off-by: kerthcet <kerthcet@gmail.com>

kerthcet committed:
Refactor code and annotations for readability

Signed-off-by: kerthcet <kerthcet@gmail.com>

kerthcet commented:
cc @alculquicondor

alculquicondor commented:
Can you add some details on what this refactor is about in the description?

Also, maybe this can wait until after code freeze and before test freeze.

alculquicondor commented on test/integration/scheduler_perf/scheduler_perf_test.go:
```diff
@@ -52,23 +52,30 @@ import (
+	sleepOpcode            operationCode = "sleep"
+)
 
+const (
```
just leaving empty lines within the `const` block should be enough, but up to you.

kerthcet replied:
Block can help to declare a group of related constants or variables, prefer to keep it.

alculquicondor commented on test/integration/scheduler_perf/scheduler_perf_test.go:
```diff
@@ -84,27 +91,31 @@ var (
-	// Default path to spec file describing the pods to create. Optional.
+	// SchedulerConfigFilePath is the path of scheduler configuration
+	// Optional
+	SchedulerConfigFilePath *string
```
```suggestion
	SchedulerConfigPath *string
```
?

kerthcet replied:
Updated and squashed.

alculquicondor reviewed (commented):
/approve

kerthcet committed:
Refactor SchedulerConfigFile

Rename to SchedulerConfigPath and make it a pointer
to be consist with other fields

Signed-off-by: kerthcet <kerthcet@gmail.com>

alculquicondor commented on test/integration/scheduler_perf/scheduler_perf_test.go:
```diff
@@ -84,27 +91,31 @@ var (
-	// Default path to spec file describing the pods to create. Optional.
+	// SchedulerConfigPath is the path of scheduler configuration
+	// Optional
+	SchedulerConfigPath *string
```
Were there no YAML files using this field?

kerthcet replied:
Yes, search the `schedulerConfigFile` with no result return.

alculquicondor commented:
/lgtm
